### PR TITLE
HBP-75 Make tvb-contrib a Python module

### PR DESCRIPTION
**Pull request by bvalean**
We consider to bring some of the code produced by @dionperd  https://github.com/dionperd/tvb-scripts under tvb "contrib".
In order to allow easy usage of this new code, we propose to make it a Python module. Before, it was just a folder with independent Python files inside. The previous form was enough for very simple unlinked extensions, but now we want a group of related and reusable modules.
How do you feel about a tvb-contrib module? Other suggestions ?

### CODE REVIEW

File: scientific_library/contrib/setup.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                 include_package_data=True,
+                 install_requires=["tvb-library"],
+                 description='A package with TVB contributed additions to the simulator, useful for scripting.',
+                 license="GPL v3",
+                 author_email='tvb.admin@thevirtualbrain.org',
+                 url='http://www.thevirtualbrain.org',
+                 download_url='https://github.com/the-virtual-brain/tvb-root',
+                 keywords='tvb brain simulator neuroscience contrib')
+
+shutil.rmtree('tvb_contrib.egg-info', True)
```
Comment (maedoc) [inline scientific_library/contrib/setup.py:48]: what's the reason for this? does this prevent people from doing
```
pip install -e .
```
or 
```
python setup.py develop
```
?

File: scientific_library/contrib/tvb/contrib/scripts/datatypes/head.py [hunk 2]
```diff
@@ -2,58 +2,65 @@
...
 from tvb.datatypes.surfaces import Surface as TVBSurface
+from tvb.simulator.plot.config import CONFIGURED
 from tvb.simulator.plot.utils import raise_value_error
 
+logger = get_logger(__name__)
 
-class Head(HasTraits):
-    logger = get_logger(__name__)
 
+class Head(HasTraits):
```
Comment (liadomide) [inline scientific_library/contrib/tvb/contrib/scripts/datatypes/head.py:40]: `HasTraits` already has a `self.log` attribute, which we could use.

Comment (dionperd) [reply scientific_library/contrib/tvb/contrib/scripts/datatypes/head.py:40]: Therefore there is no need for the logger property there.

File: scientific_library/contrib/setup.py (hunk 1 above)
Comment (liadomide) [reply scientific_library/contrib/setup.py:48]: It is meant to just cleanup the current GIT clone.  We might achieve something similar with  .gitignore.
`python setup.py develop`, I haven't tested recently the pip install.
Do you see a problem  here?

File: scientific_library/contrib/tvb/contrib/scripts/datatypes/connectivity.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# coding=utf-8
+
+from tvb.contrib.scripts.datatypes.base import BaseModel
+from tvb.datatypes.connectivity import Connectivity as TVBConnectivity
+
+
+class ConnectivityH5Field(object):
```
Comment (liadomide) [inline scientific_library/contrib/tvb/contrib/scripts/datatypes/connectivity.py:7]: Do we still need this class?

File: scientific_library/contrib/tvb/contrib/scripts/datatypes/sensors.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+from tvb.datatypes.sensors import EEG_POLYMORPHIC_IDENTITY, MEG_POLYMORPHIC_IDENTITY, \
+    INTERNAL_POLYMORPHIC_IDENTITY
+from tvb.datatypes.sensors import Sensors as TVBSensors
+from tvb.datatypes.sensors import SensorsEEG as TVBSensorsEEG
+from tvb.datatypes.sensors import SensorsInternal as TVBSensorsInternal
+from tvb.datatypes.sensors import SensorsMEG as TVBSensorsMEG
+from tvb.simulator.plot.utils import ensure_list
+
+
+class SensorTypes(Enum):
```
Comment (liadomide) [inline scientific_library/contrib/tvb/contrib/scripts/datatypes/sensors.py:19]: We should consider in the future such ENUMS as part of tvb-libray, next to our base Datatypes.
They will be generically useful

File: scientific_library/contrib/tvb/contrib/scripts/datatypes/surface.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+from tvb.datatypes.surfaces import CorticalSurface as TVBCorticalSurface
+from tvb.datatypes.surfaces import EEGCap as TVBEEGCap
+from tvb.datatypes.surfaces import FaceSurface as TVBFaceSurface
+from tvb.datatypes.surfaces import SkinAir as TVBSkinAir
+from tvb.datatypes.surfaces import SkullSkin as TVBSkullSkin
+from tvb.datatypes.surfaces import Surface as TVBSurface
+from tvb.datatypes.surfaces import WhiteMatterSurface as TVBWhiteMatterSurface
+
+
+class SurfaceH5Field(object):
```
Comment (liadomide) [inline scientific_library/contrib/tvb/contrib/scripts/datatypes/surface.py:16]: Do we need this class? I think it is not really used.

File: scientific_library/contrib/tvb/contrib/scripts/utils/time_series_utils.py [hunk 6]
```diff
@@ -0,0 +1,219 @@
+# coding=utf-8
```
Comment (liadomide) [inline scientific_library/contrib/tvb/contrib/scripts/utils/time_series_utils.py:1]: Why these functions are not methods on the TS_Service class?

File: scientific_library/contrib/tvb/contrib/scripts/datatypes/time_series_xarray.py [hunk 7]
```diff
@@ -0,0 +1,727 @@
...
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+#
+#
+
+"""
+The TimeSeries datatypes.
+
+.. moduleauthor:: Stuart A. Knock <Stuart@tvb.invalid>
```
Comment (liadomide) [inline scientific_library/contrib/tvb/contrib/scripts/datatypes/time_series_xarray.py:34]: Is it really Stuart the author here?

File: scientific_library/contrib/tvb/contrib/scripts/datatypes/time_series_xarray.py [hunk 8]
```diff
@@ -0,0 +1,727 @@
...
+
+"""
+from copy import deepcopy
+
+import numpy as np
+import xarray as xr
```
Comment (liadomide) [inline scientific_library/contrib/tvb/contrib/scripts/datatypes/time_series_xarray.py:40]: We should add `xarray ` as a dependency for `tvb-contrib`

File: scientific_library/contrib/tvb/contrib/scripts/datatypes/time_series.py [hunk 9]
```diff
@@ -0,0 +1,618 @@
...
+from tvb.datatypes.time_series import TimeSeries as TimeSeriesTVB
+from tvb.datatypes.time_series import TimeSeriesEEG as TimeSeriesEEGTVB
+from tvb.datatypes.time_series import TimeSeriesMEG as TimeSeriesMEGTVB
+from tvb.datatypes.time_series import TimeSeriesRegion as TimeSeriesRegionTVB
+from tvb.datatypes.time_series import TimeSeriesSEEG as TimeSeriesSEEGTVB
+from tvb.datatypes.time_series import TimeSeriesSurface as TimeSeriesSurfaceTVB
+from tvb.datatypes.time_series import TimeSeriesVolume as TimeSeriesVolumeTVB
+from tvb.simulator.plot.utils import ensure_list
+
+TvbProfile.set_profile(TvbProfile.LIBRARY_PROFILE)
```
Comment (liadomide) [inline scientific_library/contrib/tvb/contrib/scripts/datatypes/time_series.py:23]: Why hard-coding the profile here? I advice not to

File: scientific_library/contrib/tvb/contrib/scripts/datatypes/connectivity.py (hunk 3 above)
Comment (dionperd) [inline scientific_library/contrib/tvb/contrib/scripts/datatypes/connectivity.py:7]: No you are right. Let'e remove it.

Comment (dionperd) [reply scientific_library/contrib/tvb/contrib/scripts/datatypes/connectivity.py:7]: Similarly for SurfaceH5Field

Comment (bvalean) [reply scientific_library/contrib/tvb/contrib/scripts/datatypes/connectivity.py:7]: This class is not used anymore. I will delete it.

File: scientific_library/contrib/tvb/contrib/scripts/datatypes/sensors.py (hunk 4 above)
Comment (bvalean) [reply scientific_library/contrib/tvb/contrib/scripts/datatypes/sensors.py:19]: I added a TODO here. When we will move this enum to the library we will have to review all usages of the sensor types harcoded strings in library.

File: scientific_library/contrib/tvb/contrib/scripts/datatypes/time_series_xarray.py (hunk 7 above)
Comment (dionperd) [reply scientific_library/contrib/tvb/contrib/scripts/datatypes/time_series_xarray.py:34]: Nope...

File: scientific_library/contrib/tvb/contrib/scripts/utils/time_series_utils.py (hunk 6 above)
Comment (dionperd) [reply scientific_library/contrib/tvb/contrib/scripts/utils/time_series_utils.py:1]: To shorten ts service and to have the available for other possible uses.

File: scientific_library/contrib/tvb/contrib/scripts/datatypes/time_series_xarray.py (hunk 7 above)
Comment (liadomide) [reply scientific_library/contrib/tvb/contrib/scripts/datatypes/time_series_xarray.py:34]: Then the entire header for this file can be removed.

File: scientific_library/contrib/tvb/contrib/scripts/utils/time_series_utils.py (hunk 6 above)
Comment (liadomide) [reply scientific_library/contrib/tvb/contrib/scripts/utils/time_series_utils.py:1]: I see.... 
Just gathering a lot of functions under "utils" is not a good practice in programming, as it tends to grow without much structure (other that it is "useful" code for "some purpose", purpose which often remains unclear to the end user of that code).
We could leve these be, for the moment, but I advice to convert these in Services, where possible.

Comment (dionperd) [reply scientific_library/contrib/tvb/contrib/scripts/utils/time_series_utils.py:1]: Then, Bogdan should feel free to refactor at will inside contrib.
If I ever need to use these functions in another context, I can always copy-paste them from my own tvb-scripts repo.

Comment (dionperd) [reply scientific_library/contrib/tvb/contrib/scripts/utils/time_series_utils.py:1]: The same goes for file_utils for instance and all other utils.
Functions not used could be removed, and maybe re-added at another moment in future etc

Comment (liadomide) [reply scientific_library/contrib/tvb/contrib/scripts/utils/time_series_utils.py:1]: Good. 
Let us do this after we merge the current PR, as it already gathering many changes.

**Pull request merged**